dbms/src/Common/tests/gtest_rw_lock.cpp

```diff
@@ -123,6 +131,74 @@ TEST(Common, RWLock_Recursive)
 }
 
 
+TEST(Common, RWLock_Deadlock)
+{
+    static auto lock1 = RWLockImpl::create();
+    static auto lock2 = RWLockImpl::create();
+
+    /**
+      * q1: r1          r2
```

Comment:
Just as a note - DL detection alarm would trigger as well for just q1 and q4

Reply:
:+1: